pkgs/by-name/xo/xonsh/package.nix

```diff
@@ -1,23 +1,122 @@
-{ lib
-, callPackage
-, extraPackages ? (ps: [ ])
-, runCommand
+{
+  lib,
+  callPackage,
+  coreutils,
+  fetchFromGitHub,
+  fetchPypi,
+  git,
+  gitUpdater,
+  glibcLocales,
+  python3,
 }:
 
 let
-  xonsh-unwrapped = callPackage ./unwrapped.nix { };
-  inherit (xonsh-unwrapped.passthru) python;
+  # workaround for "Bad file descriptor in prompt_toolkit > 3.0.40"
+  # https://github.com/xonsh/xonsh/issues/5241
+  # https://github.com/prompt-toolkit/python-prompt-toolkit/issues/1853
+  prompt-toolkit' = python3.pkgs.prompt-toolkit.overridePythonAttrs (old: {
+    version = "3.0.40";
+    src = fetchPypi {
+      pname = "prompt_toolkit";
+      version = "3.0.40";
+      hash = "sha256-o3HAa7HWbNSZ/s1wjlDAtq4ArLqYIrozxYbi8W0bc54=";
+    };
+  });
+
+  argset = {
```

Comment:
does it really make sense to define the argset here and not inline it and pull the version from rec? Seems unnecessarily hard to read to first define the constant.

Reply:
No, it does not. I am just avoiding `rec` at all costs (as explained in the commit messages).

https://nix.dev/guides/best-practices.html#recursive-attribute-set-rec

Reply:
> No, it does not. I am just avoiding `rec` at all costs

Fair, red is controversial. Would the finalAttrs pattern work with buildPythonApplication or is that not a thing? I can respect avoiding rec, and if finalAttrs trick doesn't work on buildPythonApplication then i suppose this is clean already

Reply:
`finalAttrs` is not a thing yet.